sklearn/manifold/tests/test_isomap.py

```diff
@@ -250,6 +250,31 @@ def test_isomap_fit_precomputed_radius_graph():
     assert_allclose(precomputed_result, result)
 
 
+def test_isomap_fitted_attributes_dtype(global_dtype):
+    """Check that the fitted attributes are stored accordingly to the
+    data type of X."""
+    iso = manifold.Isomap(n_neighbors=2)
+
+    X = np.array([[1, 2], [3, 4], [5, 6]], dtype=global_dtype)
+
+    iso.fit(X)
+
+    assert iso.dist_matrix_.dtype == global_dtype
+
```

Comment:
`embedding_` is the only other instance attribute whose dtype must be the input's.
```suggestion
    assert iso.dist_matrix_.dtype == global_dtype
    assert iso.embedding_.dtype == global_dtype
```